### PR TITLE
creating entities

### PR DESCRIPTION
**Pull request by Stanley-Maraba**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			<artifactId>mysql-connector-j</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+
+			<dependency>
```
Comment (colsen-neslo) [inline pom.xml]: Fix formatting

File: src/main/java/com/budgetapp/entities/caregory.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.budgetapp.entities;
+
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "user")
+public class caregory {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/caregory.java]: Follow Java case standards

File: src/main/java/com/budgetapp/entities/caregory.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.budgetapp.entities;
+
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "user")
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/caregory.java]: Table name does not match class name.

File: src/main/java/com/budgetapp/entities/caregory.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package com.budgetapp.entities;
+
+import jakarta.persistence.*;
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/caregory.java]: Rather user javax persistence api.

Comment (colsen-neslo) [reply src/main/java/com/budgetapp/entities/caregory.java]: Avoid using wildcard imports and only import the exact api's as needed.

File: src/main/java/com/budgetapp/entities/expence.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package com.budgetapp.entities;
+
+
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "expence")
+public class expence {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/expence.java]:
```suggestion
@Table(name = "expense")
public class Expense {
```

File: src/main/java/com/budgetapp/entities/expence.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/expence.java]: You can create an abstract entity that your entities extend from that will contain any common attributes for your domain models.

File: src/main/java/com/budgetapp/entities/user.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package com.budgetapp.entities;
+
+import jakarta.persistence.*;
+import jakarta.persistence.criteria.CriteriaBuilder;
+
+@Entity
+@Table(name = "user")
+public class user {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/user.java]:
```suggestion
@Table(name = "user")
public class User {
```

File: src/main/java/org/example/Main.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package org.example;
+
+public class Main {
```
Comment (colsen-neslo) [inline src/main/java/org/example/Main.java]: This is not required, you are building a spring boot app.

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,8 @@
+
+spring.datasource.url=jdbc:postgresql://localhost:5432/budgetApp
+spring.datasource.username=postgres
+spring.datasource.password=10111@Connect
+spring.jpa.hibernate.ddl-auto=create
```
Comment (colsen-neslo) [inline src/main/resources/application.properties]: Make use of a property that can create and update.

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+			<scope>runtime</scope>
+			<optional>true</optional>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+				<groupId>org.postgresql</groupId>
```
Comment (JOlsen-Neslo) [inline pom.xml]: Fix your formatting in this file

File: src/main/java/com/budgetapp/BudgetappApplication.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package com.budgetapp;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class BudgetappApplication {
+
+	public static void main(String[] args) {
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/BudgetappApplication.java]: All parameters for functions should be marked as final

File: src/main/java/com/budgetapp/BudgetappApplication.java [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+		SpringApplication.run(BudgetappApplication.class, args);
+	}
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/BudgetappApplication.java]: The indent size is different across your classes, make sure your config is set and keep each file formatted the same please

File: src/main/java/com/budgetapp/entities/Category.java [hunk 13]
```diff
@@ -0,0 +1,21 @@
+package com.budgetapp.entities;
+
+
+import jakarta.persistence.Entity;
+
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "Category")
+public class Category extends BaseEntity{
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Category.java]: Please format this file

File: src/main/java/com/budgetapp/entities/Category.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Category.java]:
```suggestion
    public void setName(final String name) {
```

File: src/main/java/com/budgetapp/entities/Category.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "Category")
+public class Category extends BaseEntity{
+
+    private String name;
+
+    public String getName() {
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Category.java:12]: Up to you, but I generally like to use the Lombok library to dynamically create accessor and mutators to keep the class lean

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 16]
```diff
@@ -0,0 +1,29 @@
+package com.budgetapp.entities;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "expense")
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java]: lowercase or Pascal case is fine, but please make sure all entities follow the same standard

`Category` vs `expense`

Should be:

`categories` vs `expenses`

or 

`Categories` vs `Expenses`

File: pom.xml [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+				<artifactId>postgresql</artifactId>
```
Comment (colsen-neslo) [inline pom.xml]: Fix formatting

File: src/main/java/com/budgetapp/entities/Category.java [hunk 18]
```diff
@@ -0,0 +1,21 @@
+package com.budgetapp.entities;
+
+
+import jakarta.persistence.Entity;
+
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "Category")
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Category.java]: Table names should be in plural form

```suggestion
@Table(name = "Categories")
```

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+public class Expense extends BaseEntity{
+
+
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java:11]: Make sure there aren't any trailing blank lines across your app where its not needed. A tool like Checkstyle could help enforce this at compile time.

Also, please format this class

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+    private String name;
+    private Integer value;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java]:
```suggestion
    public void setName(final String name) {
```

File: src/main/java/com/budgetapp/entities/User.java [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+@Entity
+@Table(name = "users")
+public class User extends BaseEntity{
+
+
+    private String username;
+    private String password;
+    private String roles;
+
+    public String getName() {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/User.java]: Make sure your getters and setters are named correctly to the field.
```suggestion
    public String getUsername() {
```

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+        this.name = name;
+    }
+
+    public Integer getValue() {
+        return value;
+    }
+
+    public void setValue(Integer value) {
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java]:
```suggestion
    public void setValue(final Integer value) {
```

File: src/main/java/com/budgetapp/entities/User.java [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+        return username;
+    }
+
+    public void setName(String name) {
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/User.java]:
```suggestion
    public void setUsername(String name) {
```

File: src/main/java/org/example/Main.java (hunk 8 above)
Comment (colsen-neslo) [reply src/main/java/org/example/Main.java]: This hasn't been removed.

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "expense")
+public class Expense extends BaseEntity{
+
+
+    private String name;
+    private Integer value;
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java]: When dealing with monetary values, its generally good practice to use `BigDecimal` so you can cater for cent values

File: src/main/java/com/budgetapp/entities/User.java [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "users")
+public class User extends BaseEntity{
+
+
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/User.java:12]: Same comment here regarding blank lines and formatting

File: src/main/java/com/budgetapp/entities/User.java (hunk 23 above)
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/User.java]: Mark all parameters as final

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]
Nice first attempt! Left some comments regarding the style of code we expect. Let me know if anything is unclear so we can chat about it.

Also, please make sure each PR that you create is based on a ticket on the JIRA board. The branch name should be in this format: `feature/SBA-{ticket-number}`

Then make sure each commit is prefixed with the issue code and number: `SBA-123: create entities`

File: src/main/java/com/budgetapp/entities/Category.java (hunk 13 above)
Comment (colsen-neslo) [reply src/main/java/com/budgetapp/entities/Category.java]: This is still outstanding.

File: src/main/java/com/budgetapp/entities/User.java (hunk 21 above)
Comment (colsen-neslo) [reply src/main/java/com/budgetapp/entities/User.java]: This still needs to be done

File: src/main/java/com/budgetapp/entities/User.java [hunk 26]
```diff
@@ -1,17 +1,17 @@
 package com.budgetapp.entities;
 
-
 import jakarta.persistence.Entity;
 import jakarta.persistence.Table;
+import java.math.BigDecimal;
 
 @Entity
 @Table(name = "users")
 public class User extends BaseEntity{
```
Comment (colsen-neslo) [inline src/main/java/com/budgetapp/entities/User.java]: Please use your IDE to format all your classes (`cmd + option + L`)

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

**Review by colsen-neslo — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (JOlsen-Neslo) [inline pom.xml]: This formatting is still off

File: src/main/java/com/budgetapp/entities/Expense.java [hunk 27]
```diff
@@ -0,0 +1,30 @@
+package com.budgetapp.entities;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+import java.math.BigDecimal;
+
+@Entity
+@Table(name = "expenses")
```
Comment (JOlsen-Neslo) [inline src/main/java/com/budgetapp/entities/Expense.java:9]:
```suggestion
@Table(name = "Expenses")
```

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

**Review by JOlsen-Neslo — APPROVED** [review]

**Pull request merged**